scipy/_lib/_array_api.py

```diff
@@ -163,3 +163,53 @@ def copy(x, *, xp=None):
         xp = array_namespace(x)
 
     return as_xparray(x, copy=True, xp=xp)
+
+
+def assert_equal(actual, desired, err_msg='', xp=None):
+    # TODO compare arrays from different libraries
+    if xp is None:
+        xp = array_namespace(actual)
+    if 'cupy' in xp.__name__:
+        return xp.testing.assert_array_equal(actual, desired, err_msg=err_msg)
+    elif 'torch' in xp.__name__:
+        return xp.testing.assert_close(actual, desired, rtol=0, atol=0,
+                                       msg=err_msg)
+    return np.testing.assert_array_equal(actual, desired, err_msg=err_msg)
+
+
+def assert_close(actual, desired, rtol=None, atol=None, err_msg='', xp=None):
+    # TODO compare arrays from different libraries
+    if xp is None:
+        xp = array_namespace(actual)
+    if 'cupy' in xp.__name__:
+        if rtol is None:
+            rtol = 1e-07
+        if atol is None:
+            atol = 0
+        return xp.testing.assert_allclose(actual, desired, rtol=rtol,
+                                          atol=atol, err_msg=err_msg)
+    elif 'torch' in xp.__name__:
+        return xp.testing.assert_close(actual, desired, rtol=rtol,
+                                       atol=atol, msg=err_msg)
```

Comment:
The `atol` defaults for `torch.testing.assert_close` are not great. How about using an explicit `atol=0` for everything in this unified testing function?

Reply:
The defaults for every library are now `rtol=1e07, atol=0`.

Reply:
LGTM, resolving